tools/isobuild/compiler-plugin.js

```diff
@@ -1394,8 +1397,19 @@ export class PackageSourceBatch {
     if (canCache) {
       LINKER_CACHE.set(cacheKey, ret);
       if (cacheFilename) {
+        const cacheKeySuffixRegex = new RegExp('_' + cacheKeySuffix, 'g');
+        const oldCacheFilename =
+          cacheFilename.replace(cacheKeySuffixRegex, '_*');
         // Write asynchronously.
-        Fiber(() => files.writeFileAtomically(cacheFilename, retAsJSON)).run();
+        Fiber(() => {
+          try {
+            files.rm_recursive(oldCacheFilename);
```

Comment:
We could come up with another way to handle this for sure - I'm just not sure if it's worth adding extra complexity though, since the delete + write are running in a fiber anyways (and we’re not worried about having the new cache file available until the next rebuild). Happy to rework any of this though!